File: examples/python/operations/dag/dag-signature.py
from aws_durable_execution_sdk_python import (
    DagContext,
    DagResult,
    DurableContext,
    durable_execution,
)
@durable_execution
def handler(event: dict, context: DurableContext) -> int:
    def register(dag: DagContext) -> None:
        dag.step(lambda deps, sc: 42, name="double")

    result: DagResult = context.dag(register, name="compute")
    return result.get_result("double")

File: examples/java/operations/dag/dag-signature.java
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;
import software.amazon.lambda.durable.dag.DagResult;

public class DagSignature extends DurableHandler<Void, Integer> {
    @Override
    public Integer handleRequest(Void input, DurableContext context) {
        DagResult result = context.dag("compute", dag -> {
            dag.step("double", Integer.class, (deps, ctx) -> 42);
        });

        return (Integer) result.getResult("double").orElseThrow();
    }
}
